model-client/src/commonMain/kotlin/org/modelix/model/client2/IModelClientV2.kt

```diff
@@ -13,14 +13,13 @@
  */
 package org.modelix.model.client2
 
-import io.ktor.utils.io.core.Closeable
 import org.modelix.model.IVersion
 import org.modelix.model.api.IIdGenerator
 import org.modelix.model.lazy.BranchReference
 import org.modelix.model.lazy.RepositoryId
 import org.modelix.model.server.api.ModelQuery
 
-interface IModelClientV2 : Closeable {
```

Comment:
Shouldn't this be implemented instead: https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-auto-closeable/

Reply:
in the JVM implementation then

Reply:
https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-auto-closeable/ is `@ExperimentalStdlibApi`, though I avoided it.

Also, deriving a JVM specific implementation to have it implement https://docs.oracle.com/javase/8/docs/api/java/io/Closeable.html made not too much sense.

Reply:
As far as I can tell, this change is now breaking for JVM customers. The ktor [Closeable](https://api.ktor.io/ktor-io/io.ktor.utils.io.core/-closeable/index.html) as an implementation of JVM closeable. Thus, in the JVM world, IModelClientV2 could be used with `use`. Without providing some means of implementing Closeable in JVM again, using the client inside `use` or try-with-resource in Java is not possible anymore now. Therefore, this would need to be marked as a breaking change.

Reply:
Ok, I see. I will add something, that does not break the JVM implementation.

Reply:
You can have a look at it again.

Reply:
Wouldn't it be less code to simple do the same trick that ktor does with their Closeable: declare that as a platform specific interface and implement it differently per platform. Then, you wouldn't have to rebuild the whole IModelClientV2 interface in each platform.

Reply:
But then we would have something like `org.modelix.core.Closable` which would become part of our API and if we were to remove or move it, it would be a breaking change. For now, I would like to avoid a modelix-specific `Closable`-interface.

Reply:
This seems to do the trick.

```
package org.modelix.model.client2

internal expect interface Closable {
    fun close()
}
```

Because the interface is internal, it does not get exposed. 
But still Javas `java.io.Closeable`, `java.io.Closeable.close` and the `use`-extension functions work.

Reply:
The current solution looks fine. A lot less code to maintain now.